### PR TITLE
Make some things a little saner

### PR DESCRIPTION
**Pull request by fractalcat**
- [x] Don't guess static files by directory name, they should be data-files in the .cabal.
- [x] Preserve mode, ownership, et cetera when copying stuff during RPM builds.
- [x] Don't flatten directories when copying static files under Debian.
- [x] Don't flatten directories when copying static files under Centos.

### CODE REVIEW

File: lib/Anchor/Package/Process.hs [hunk 1]
```diff
@@ -237,3 +231,15 @@ extractCabalDetails fp = do
              (maintainer  pd)
              (map fst $ condExecutables gpd)
              (dataFiles   pd)
+
+-- | Copy a data-file (possibly deep within the repository) to the
+--   equivalent place in the provided destination tree, creating it if
+--   it doesn't exist.
+--
+--   Calls `bash -c` to deal with globbing; if you call this on
+--   untrusted input, you wil be fired, and then you will die.
```
Comment (oswynb) [inline lib/Anchor/Package/Process.hs]: s/wil/will

File: lib/Anchor/Package/Template.hs [hunk 2]
```diff
@@ -63,16 +64,15 @@ generateM4 = do
                , "dnl"
                ]
     generateCopyStrings :: [String] -> [String] -> String
-    generateCopyStrings execs datas = executableStrings execs <> dataStrings datas
+    generateCopyStrings execs datas = executableStrings execs <> generateDataCopyStrings datas
       where
-        executableStrings = unlines . map (\x -> "cp -v dist/build/" <> x <> "/" <> x <> " %{buildroot}%{_bindir}")
-        dataStrings = unlines . map (\x -> "cp -v " <> x <> "/" <> x <> " %{buildroot}%{_datadir}")
+        executableStrings = unlines . map (\x -> "cp -va dist/build/" <> x <> "/" <> x <> " %{buildroot}%{_bindir}")
```
Comment (oswynb) [inline lib/Anchor/Package/Template.hs]: While things changing should probably change the `x <> "/" <> x` to `x </> x`

File: lib/Anchor/Package/Template.hs [hunk 3]
```diff
@@ -63,16 +64,15 @@ generateM4 = do
...
 
     generateFileStrings :: [String] -> [String] -> String
     generateFileStrings execs datas = executableStrings execs <> dataStrings datas
       where
         executableStrings = unlines . map (\x -> "%{_bindir}/" <> x)
```
Comment (oswynb) [inline lib/Anchor/Package/Template.hs]: </> here too

File: lib/Anchor/Package/Process.hs (hunk 1 above)
Comment (fractalcat) [inline lib/Anchor/Package/Process.hs]: Fixed.

File: lib/Anchor/Package/Template.hs (hunk 2 above)
Comment (fractalcat) [inline lib/Anchor/Package/Template.hs]: Fixed.

File: lib/Anchor/Package/Template.hs (hunk 3 above)
Comment (fractalcat) [inline lib/Anchor/Package/Template.hs]: Yep.

**Pull request merged**